### ISSUE TITLE
File was not found because wrong user id

### ISSUE THREAD

**Issue by yharroch** [R0Wi/workflow_ocr#110]
Hi there,
thanks a lot for your work. It'll be heaven when I'll be able to scan to PDF with 'fulltextsearch'.
I installed 1.23.2 version on nextcloud 23.0.3 but i have a "Could not process file '/USER_1/files/_SCAN/CCF01042022_0001_OCR.pdf'. File was not found".

I think the problem is that the real owner of the file is another user. The processed file is in a shared folder.
Workflow_ocr look for the current user instead of the owner. The workflow cannot proceed because it look in the wrong data folder. Could you fix this switching 'user' to 'owner' ?

**Comment by R0Wi** [comment]
So the problem occures if you share a folder via share link and some "unknown" user uploads a PDF-file matching your workflow? Is that correct so far?

I think we already use the owner of the file (see https://github.com/R0Wi/workflow_ocr/blob/5aa118e02a37dbd48a7fcc8b97441a98dde0437e/lib/Operation.php#L125) to impersonate when processing the file asynchronously. 

I just created a folder link and uploaded a PDF file which was processed correctly by the app. So i think we need some more information here. Could you please try to temporarily set your [loglevel](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/logging_configuration.html) to `DEBUG` (0), reproduce the process and share your server logs with us?

**Comment by yharroch** [comment]
Yes. I thought the same when i looked for it in the source code. I think it comes from the getUID function that looks for the last modification user but not the real owner.

To clarify, i've a folder named '_scans' shared by **'admin_user'**. This folder is the main folder where my scanned files are saved. It means whether you're **'user_1'** or **'user_2'**, every scanned files are sent in this shared folder. 
So, when **user_1**  or **user_2** scan and create a new file, it goes in the **'admin_user'** data folder which directory is /nextcloud/data/**'admin_user'**/files/_scans/.
But [workflow_ocr] look for a file in /nextcloud/data/**'user_1'**/files/_scans/ because **'user_1'** was the one who scanned and created the file. Thus, this new file cannot be found because it is not in this directory.

```
[workflow_ocr] Warning: Could not process file '/_'user_1'_/files/_SCAN/CCF01042022_0001_OCR.pdf'. File was not found
by _admin_user_ at 2022-04-02T15:04:07+00:00
```
_I changed usernames for security reasons but this logs is this only one I've got when the process runs_

**Comment by yharroch** [comment]
FYI, I tried to run the process in another folder owned by 'user_1'. It worked like a charm ! 
I was really happy and impressed. Well done for the App !

**Comment by yharroch** [comment]
Yes. Exactly. 
Shared internally on the same nextcloud server. 
Don't shared in a group but with 2 users.

**Comment by R0Wi** [comment]
Didn't have the time to dig deeper but i'll keep you up to date 👍

**Comment by R0Wi** [comment]
@yharroch the user id is now determined by the file path so i think this should be consistent. https://github.com/R0Wi/workflow_ocr/pull/116 tackles this issue. It would be great if you could give us some feedback by installing the temporary app artifact from https://github.com/R0Wi/workflow_ocr/suites/6415613489/artifacts/234051766. Please let us now if you need further assistance. Thank's for your help!

**Comment by R0Wi** [comment]
Included in `v1.24.1` and `v1.23.3` 😎
